### PR TITLE
Remove references to deprecated CompilationUnit, upstream changes.

### PR DESCRIPTION
**Pull request by raghav-deepsource**
This PR removes all references to CompilationUnit from Spoon. it also integrates changes from INRIA/spoon master into our fork.

### CODE REVIEW

**Comment by raghav-deepsource** [comment]
@devashish-deepsource , please approve.

File: src/test/java/spoon/test/targeted/TargetedExpressionTest.java [hunk 1]
```diff
@@ -401,13 +404,29 @@ public void testTargetsOfInvInInnerClass() throws Exception {
 		final List<CtInvocation<?>> elements = innerInvMethod.getElements(new TypeFilter<>(CtInvocation.class));
 		assertEquals(8, elements.size());
 		expectedThisAccess.setType(expectedInnerClass);
-		assertEqualsInvocation(new ExpectedTargetedExpression().declaringType(expectedType).target(expectedThisAccess).result("inv()"), elements.get(0));
+		assertThat(elements.get(0).getTarget().isImplicit(), is(true));
+		assertThat(elements.get(0).getTarget(), is(instanceOf(CtThisAccess.class)));
+		assertThat(
+			((CtTypeAccess<?>) ((CtThisAccess<?>) elements.get(0).getTarget()).getTarget())
+				.getAccessedType()
+				.getQualifiedName(),
```
Comment (devashish-deepsource) [inline src/test/java/spoon/test/targeted/TargetedExpressionTest.java:412]: If chains get complicated, consider introducing new variables for better readability.

File: src/test/java/spoon/test/targeted/TargetedExpressionTest.java [hunk 2]
```diff
@@ -401,13 +404,29 @@ public void testTargetsOfInvInInnerClass() throws Exception {
...
 		expectedSuperThisAccess.setType(expectedInnerClass);
-		assertEqualsInvocation(new ExpectedTargetedExpression().declaringType(expectedSuperClassType).target(expectedSuperThisAccess).result("superMethod()"), elements.get(4));
+		assertThat(elements.get(4).getTarget().isImplicit(), is(true));
+		assertThat(elements.get(4).getTarget(), is(instanceOf(CtThisAccess.class)));
+		assertThat(
+			((CtTypeAccess<?>) ((CtThisAccess<?>) elements.get(4).getTarget()).getTarget())
+				.getAccessedType()
+				.getQualifiedName(),
+			is("spoon.test.targeted.testclasses.Foo")
+		);
```
Comment (devashish-deepsource) [inline src/test/java/spoon/test/targeted/TargetedExpressionTest.java:428]: If chains get complicated, consider introducing new variables for better readability.

**Review by devashish-deepsource — COMMENTED** [review]
Left some comments.

File: src/main/java/spoon/support/StandardEnvironment.java [hunk 3]
```diff
@@ -256,18 +259,32 @@ public void report(Processor<?> processor, Level level, CtElement element, Strin
 		if (sp == null) {
 			buffer.append(" (Unknown Source)");
 		} else {
-			// TODO: will explode if type == null
-			buffer.append(" at " + type.getQualifiedName() + ".");
+			if (type != null) {
+				buffer.append(" at ")
+							.append(type.getQualifiedName())
+							.append(".");
```
[the remark this answers is not in the document]
Comment (raghav-deepsource) [reply src/main/java/spoon/support/StandardEnvironment.java:265]: no clue, this is an upstream change.

File: src/main/java/spoon/support/StandardEnvironment.java [hunk 4]
```diff
@@ -256,18 +259,32 @@ public void report(Processor<?> processor, Level level, CtElement element, Strin
...
-			buffer.append("(" + sp.getFile().getName() + ":" + sp.getLine() + ")");
+			if (sp.getFile() != null) {
+				buffer.append("(")
+							.append(sp.getFile().getName())
+							.append(":")
+							.append(sp.getLine())
+							.append(")");
+			} else {
+				buffer.append("(?:?)");
+			}
```
[the remark this answers is not in the document]
Comment (raghav-deepsource) [reply src/main/java/spoon/support/StandardEnvironment.java:281]: upstream change.

File: src/test/java/spoon/test/targeted/TargetedExpressionTest.java (hunk 1 above)
Comment (raghav-deepsource) [reply src/test/java/spoon/test/targeted/TargetedExpressionTest.java:412]: upstream change

File: src/test/java/spoon/test/targeted/TargetedExpressionTest.java (hunk 2 above)
Comment (raghav-deepsource) [reply src/test/java/spoon/test/targeted/TargetedExpressionTest.java:428]: same

**Review by devashish-deepsource — APPROVED** [review]
LGTM.

**Review by srijan-deepsource — APPROVED** [review]

**Comment by raghav-deepsource** [comment]
Next marvin-java release will use new spoon code

**Pull request merged**